### PR TITLE
[docs] add modin_xgboost and dask_xgboost notebook tutorials

### PR DESCRIPTION
**Pull request by matthewdeng**
## Overview

This PR adds a `modin_xgboost.py`and `dask_xgboost.py` notebook that can be run by users to execute an end-to-end distributed training pipeline using XGBoost-Ray with Modin or XGBoost-Ray with Dask. The logic for these example come from [modin_xgboost_test.py](https://github.com/ray-project/ray/blob/master/release/golden_notebook_tests/workloads/modin_xgboost_test.py) and [dask_xgboost_test.py](https://github.com/ray-project/ray/blob/master/release/golden_notebook_tests/workloads/dask_xgboost_test.py).

Artifacts of this PR include:
1. Executable `.py` script.
2. Generated `.rst` file that is rendered in the documentation.
3. Downloadable `.ipynb` file that can be run directly on a cluster.
4. CI smoke test for the `.py` script. (currently disabled due to python 3.6 compatibility issues)
5. Release tests in `release/golden_notebook_tests` that run the executable `.py` script on an Anyscale cluster.

## Usage

This notebook is [sphinx-gallery](https://sphinx-gallery.github.io/stable/index.html) compatible and will be displayed in [Ray Tutorials and Examples](https://docs.ray.io/en/master/auto_examples/overview.html). 

The user can run the end-to-end notebook through the following steps (which are documented in the file):
1. Deploy an OSS cluster using the `modin_xgboost.yaml` cluster config.
2. Download `modin_xgboost.ipynb` and move it onto the cluster.
3. Run Jupyter Notebook on the cluster and access it from their local browser.
4. Run the job by executing the code in the `modin_xgboost.ipynb`.  

Same can be done for `dask_xgboost`.

**Notes:** 
1. Running `ray up` to deploy the cluster requires AWS credentials to be set. These have been omitted in the notebook, but should be configured by the user.
2. This notebook currently uses OSS Ray, but can be run on Anyscale by modifying the cluster setup steps.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by matthewdeng** [comment]
@scv119 here's the PR. The python script is integrated with CI and also converts into a notebook that can be directly run on a cluster.

Generated doc is [here](https://ray--18775.org.readthedocs.build/en/18775/auto_examples/modin_xgboost/modin_xgboost.html).

File: doc/examples/modin_xgboost/modin_xgboost.py [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+###############################################################################
+#  Override these arguments as needed:
+
+address = args.address
+smoke_test = args.smoke_test
+num_actors = args.num_actors
+cpus_per_actor = args.cpus_per_actor
+num_actors_inference = args.num_actors_inference
+cpus_per_actor_inference = args.cpus_per_actor_inference
```
[the remark this answers is not in the document]
Comment (matthewdeng) [reply doc/examples/modin_xgboost/modin_xgboost.py:101]: The would not be set. In the Jupyter notebook, the previous cell will be run and parse the args  for `ipykernel_launcher`, which will just take all the default values.

[the remark this answers is not in the document]
Comment (Yard1) [reply doc/examples/modin_xgboost/modin_xgboost.py:101]: Ah, gotcha

**Review by Yard1 — APPROVED** [review]

File: doc/examples/dask_xgboost/dask_xgboost.py [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+XGBoost-Ray with Dask
+======================
+
+This notebook includes an example workflow using
+`XGBoost-Ray <https://docs.ray.io/en/latest/xgboost-ray.html>`_ and
+`Dask <https://docs.dask.org/en/latest/>`_ for distributed model training,
+hyperparameter optimization, and prediction.
+"""
+
+###############################################################################
```
Comment (Yard1) [inline doc/examples/dask_xgboost/dask_xgboost.py:12]:
```suggestion
"""

# If you are moving this file, please also update the path in
# release/golden_notebook_tests/workloads/dask_xgboost_test.py

###############################################################################
```

File: doc/examples/modin_xgboost/modin_xgboost.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+XGBoost-Ray with Modin
+======================
+
+This notebook includes an example workflow using
+`XGBoost-Ray <https://docs.ray.io/en/latest/xgboost-ray.html>`_ and
+`Modin <https://modin.readthedocs.io/en/latest/>`_ for distributed model
+training and prediction.
+"""
+
+###############################################################################
```
Comment (Yard1) [inline doc/examples/modin_xgboost/modin_xgboost.py:11]:
```suggestion
"""

# If you are moving this file, please also update the path in
# release/golden_notebook_tests/workloads/modin_xgboost_test.py

###############################################################################
```

**Review by Yard1 — COMMENTED** [review]
@matthewdeng Added suggestions for comments regarding keeping the path in release test scripts up to date, please take a look!

File: doc/examples/dask_xgboost/dask_xgboost.py (hunk 2 above)
Comment (matthewdeng) [reply doc/examples/dask_xgboost/dask_xgboost.py:12]: I would prefer to move this somewhere else (maybe the README?) as this is not relevant to the users who are the primary readers of the docs/script.

Comment (Yard1) [reply doc/examples/dask_xgboost/dask_xgboost.py:12]: Yeah, I wasn't 100% sure where to put this in myself. I can see the case for README, but on the other hand, I feel information like this should be directly in the file it concerns. That being said, having it anywhere is better than nowhere!

**Pull request merged**